Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -423,8 +446,386 @@ theorem sum_congr {f : SkewMonoidAlgebra k G} {M : Type*} [AddCommMonoid M] {g
     (h : ∀ x ∈ f.support, g₁ x (f.coeff x) = g₂ x (f.coeff x)) :
     f.sum g₁ = f.sum g₂ := Finset.sum_congr rfl h
 
+theorem induction_on {p : SkewMonoidAlgebra k G → Prop} (f : SkewMonoidAlgebra k G)
+    (h0 : p 0) (hM : ∀ g a, p (single g a)) (hadd : ∀ f g :
+    SkewMonoidAlgebra k G, p f → p g → p (f + g)) : p f := by
+  rw [← sum_single f, sum_def']
+  exact Finset.sum_induction _ _ hadd h0 (by aesop)
+
+/--
+Slightly less general but more convenient version of `SkewMonoidAlgebra.induction_on`. -/
+theorem induction_on' [instNonempty : Nonempty G] {p : SkewMonoidAlgebra k G → Prop}
+    (f : SkewMonoidAlgebra k G) (hM : ∀ g a, p (single g a)) (hadd : ∀ f g :
+    SkewMonoidAlgebra k G, p f → p g → p (f + g)) : p f :=
+  induction_on _ (by simpa using hM (Classical.choice instNonempty) 0) hM hadd
+
 end sum
 
+section mapDomain
```

Comment:
That's not important, but I wonder whether there would be a more obvious name than `mapDomain` for this functoriality property.